src/OrleansRuntime/Hosting/DefaultSiloServices.cs

```diff
@@ -184,13 +184,12 @@ internal static void AddDefaultServices(IServiceCollection services)
                     return new ConsistentRingProvider(siloDetails.SiloAddress);
                 });
             
-            services.AddSingleton(typeof(IKeyedServiceCollection<,>), typeof(KeyedServiceCollection<,>));
+            services.TryAddSingleton(typeof(IKeyedServiceCollection<,>), typeof(KeyedServiceCollection<,>));
 
             // Transactions
-            services.AddSingleton<ITransactionAgent,TransactionAgent>();
-            services.AddSingleton(sp => new Lazy<ITransactionAgent>(() => sp.GetRequiredService<ITransactionAgent>()));
-            services.AddSingleton<TransactionServiceGrainFactory>();
-            services.AddSingleton(sp => sp.GetRequiredService<TransactionServiceGrainFactory>().CreateTransactionManagerService() );
+            services.TryAddSingleton<ITransactionAgent,TransactionAgent>();
+            services.TryAddSingleton(sp => new Lazy<ITransactionAgent>(() => sp.GetRequiredService<ITransactionAgent>()));
```

Comment:
are you trying to register a Lazy? if that's the case, please use the Factory delegate instead, as Lazy is non-conforming, and some containers might resolve it differently